horovod/mxnet/CMakeLists.txt

```diff
@@ -6,7 +6,7 @@ set(Mxnet_TARGET_LIB "mxnet")
 
 # Find MXNet
 set(Mxnet_REQUIRED "")
-if ("$ENV{HOROVOD_WITH_MXNET}" STREQUAL "1")
+if ("$ENV{HOROVOD_WITH_MXNET}" MATCHES "^1\.")
     set(Mxnet_REQUIRED "REQUIRED")
 endif ()
 find_package(Mxnet "1.4.1" ${Mxnet_REQUIRED})
```

Comment:
Shouldn't the `$ENV{HOROVOD_WITH_MXNET}` version be also used in find_package?

Reply:
Since it's not `find_package(Mxnet "1.4.1" EXACT ${Mxnet_REQUIRED})` it still works as the package found just has to be compatible with "1.4.1" which in local testing using "1.9.1" was true - or at least all the tests/assertions/checks passed when working with it. 

I've opened a discussion topic on whether Mxnet should be supported in future since it's been moved to the Apache attic effectively meaning it's not being actively developed anymore. #3990